ddl/index.go

```diff
@@ -815,32 +832,32 @@ func (w *worker) handleReorgTasks(startTime time.Time, startHandle int64, reorgI
 func (w *worker) buildKVRangesIndex(t table.Table, workers []*addIndexWorker, kvRanges []kv.KeyRange, job *model.Job, reorgInfo *reorgInfo) error {
 	var (
 		startTime   time.Time
-		startHandle int64
-		endHandle   int64
-		err         error
+		firstHandle int64
 	)
 	totalAddedCount := job.GetRowCount()
 	batchTasks := make([]*reorgIndexTask, 0, len(workers))
 
 	log.Infof("[ddl-reorg] start to reorg index of %v region ranges.", len(kvRanges))
 	for i, keyRange := range kvRanges {
 		startTime = time.Now()
-
-		startHandle, endHandle, err = decodeHandleRange(keyRange)
+		startHandle, endHandle, err := decodeHandleRange(keyRange)
 		if err != nil {
 			return errors.Trace(err)
 		}
+		if i == 0 {
+			firstHandle = startHandle
+		}
 		endKey := t.RecordKey(endHandle)
 		endIncluded := false
 		if endKey.Cmp(keyRange.EndKey) < 0 {
 			endIncluded = true
 		}
-		task := &reorgIndexTask{startHandle, endHandle, endIncluded}
+		task := &reorgIndexTask{startHandle, endHandle, endIncluded, false}
 
 		batchTasks = append(batchTasks, task)
 		if len(batchTasks) >= len(workers) || i == (len(kvRanges)-1) {
 			// Wait tasks finish.
-			err = w.handleReorgTasks(startTime, startHandle, reorgInfo, &totalAddedCount, workers, batchTasks)
+			err = w.handleReorgTasks(startTime, firstHandle, reorgInfo, &totalAddedCount, workers, batchTasks)
```

Comment:
I guess the `firstHandle` argument is redundant.
You can always get it from `batchTasks[0].start`, right?